contracts/test/StakeTable.t.sol

```diff
@@ -0,0 +1,272 @@
+            EdOnBN254.EdOnBN254Point(schnorrVKx, schnorrVKy), // schnorrVK
+            BN254.G1Point(blsSigX, blsSigY) // sig
+        );
+    }
+
+    function setUp() public {
+        exampleTokenCreator = makeAddr("tokenCreator");
+        vm.prank(exampleTokenCreator);
+        token = new ExampleToken(INITIAL_BALANCE);
+
+        LightClient.LightClientState memory genesis = LightClient.LightClientState({
+            viewNum: 0,
+            blockHeight: 0,
+            blockCommRoot: 0,
+            feeLedgerComm: 0,
+            stakeTableBlsKeyComm: 0,
+            stakeTableSchnorrKeyComm: 0,
+            stakeTableAmountComm: 0,
+            threshold: 0
+        });
+        lightClientContract = new LightClient(genesis,10);
+        address lightClientAddress = address(lightClientContract);
+        stakeTable = new S(address(token),lightClientAddress);
+    }
+
+    function test_RevertWhen_UsingRestakeToken() external {
+        uint64 depositAmount = 10;
+        uint64 validUntilEpoch = 5;
+
+        (
+            BN254.G2Point memory blsVK,
+            EdOnBN254.EdOnBN254Point memory schnorrVK,
+            BN254.G1Point memory sig
+        ) = genClientWallet(exampleTokenCreator);
+
+        // Throw "Restaking not implemented" error
+        vm.expectRevert(S.RestakingNotImplemented.selector);
+        stakeTable.register(
+            blsVK,
+            schnorrVK,
+            depositAmount,
+            AbstractStakeTable.StakeType.Restake,
+            sig,
+            validUntilEpoch
+        );
+    }
+
+    function test_RevertWhen_InvalidBLSSig() external {
+        uint64 depositAmount = 10;
+        uint64 validUntilEpoch = 5;
+
+        (BN254.G2Point memory blsVK, EdOnBN254.EdOnBN254Point memory schnorrVK,) =
+            genClientWallet(exampleTokenCreator);
+
+        // Failed signature verification
+        BN254.G1Point memory badSig = BN254.P1();
+        vm.expectRevert(BLSSig.BLSSigVerificationFailed.selector);
+        stakeTable.register(
+            blsVK,
+            schnorrVK,
+            depositAmount,
+            AbstractStakeTable.StakeType.Native,
+            badSig,
+            validUntilEpoch
+        );
+    }
+
+    function testFuzz_RevertWhen_InvalidNextRegistrationEpoch(uint64 validUntilEpoch) external {
+        uint64 blockNumber = 20000;
+        vm.roll(blockNumber);
+
+        uint64 currentEpoch = stakeTable.currentEpoch();
+        validUntilEpoch = uint64(bound(validUntilEpoch, 0, type(uint64).max));
+        uint64 depositAmount = 10;
+        vm.prank(exampleTokenCreator);
+        token.approve(address(stakeTable), depositAmount);
+
+        (
+            BN254.G2Point memory blsVK,
+            EdOnBN254.EdOnBN254Point memory schnorrVK,
+            BN254.G1Point memory sig
+        ) = genClientWallet(exampleTokenCreator);
+
+        // Invalid next registration epoch
+        if (validUntilEpoch < currentEpoch) {
```

Comment:
to follow-up on our call, 
one thing for sure is we should remove `blockNumber` variable, since it's not doing anything.

as for `curEpoch`, we can try the following in order:
- the [`StdStorage`](https://book.getfoundry.sh/reference/forge-std/std-storage) api seems reasonable to mutate the value of `currentEpoch`
- if you couldn't get it work, then, make a `test/mocks/LightClientHelper.sol` that inherits and expose storage mutation function (see optimism's [example here](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/test/mocks/NextImpl.sol))

Reply:
Done in 9261d5b.

Reply:
I rename it so that it ends with `*Test.sol` to avoid `forge bind` to generate unnecessary bindings for it. 
[968b4a8](https://github.com/EspressoSystems/espresso-sequencer/pull/769/commits/968b4a84cd1e34b3b8df2a2528094947e19f7599)